diff --git a/liblttng-ust-python-agent/Makefile.am b/liblttng-ust-python-agent/Makefile.am
index f80b8a3..a4aeecc 100644
--- a/liblttng-ust-python-agent/Makefile.am
+++ b/liblttng-ust-python-agent/Makefile.am
@@ -16,9 +16,9 @@ all-local:
 
 install-exec-local:
 	if [ "$(DESTDIR)" = "" ]; then \
-		$(PYTHON) setup.py install --prefix=$(prefix) --record install_files.txt  --verbose --no-compile $(DISTSETUPOPTS);  \
+		$(PYTHON) setup.py install --prefix=$(prefix) --record install_files.txt --verbose --no-compile $(DISTSETUPOPTS); \
 	else \
-		$(PYTHON) setup.py install --root=$(DESTDIR) --verbose install_files.txt  --prefix=$(prefix) --no-compile $(DISTSETUPOPTS); \
+		$(PYTHON) setup.py install --prefix=$(prefix) --record install_files.txt --verbose --no-compile --root=$(DESTDIR) $(DISTSETUPOPTS); \
 	fi
 clean-local:
 	rm -rf build
