--- a/liblttng-ust-java/Makefile.am
+++ b/liblttng-ust-java/Makefile.am
@@ -1,4 +1,5 @@
 JAVAROOT = .
+AM_JAVACFLAGS = -source 1.6 -target 1.6
 jarfile = liblttng-ust-java.jar
 jardir = $(datadir)/java
 pkgpath = org/lttng/ust
@@ -8,7 +9,8 @@
 BUILT_SOURCES = org_lttng_ust_LTTngUst.h
 
 AM_CPPFLAGS = -I$(top_srcdir)/include
-lib_LTLIBRARIES = liblttng-ust-java.la
+jnilib_LTLIBRARIES = liblttng-ust-java.la
+jnilibdir = $(libdir)/jni
 liblttng_ust_java_la_SOURCES = LTTngUst.c lttng_ust_java.h
 nodist_liblttng_ust_java_la_SOURCES = org_lttng_ust_LTTngUst.h
 
--- a/liblttng-ust-java-agent/jni/jul/Makefile.am
+++ b/liblttng-ust-java-agent/jni/jul/Makefile.am
@@ -1,6 +1,7 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
-lib_LTLIBRARIES = liblttng-ust-jul-jni.la
+jnilib_LTLIBRARIES = liblttng-ust-jul-jni.la
+jnilibdir = $(libdir)/jni
 liblttng_ust_jul_jni_la_SOURCES = lttng_ust_jul.c \
 				  lttng_ust_jul.h
 
--- a/liblttng-ust-java-agent/jni/log4j/Makefile.am
+++ b/liblttng-ust-java-agent/jni/log4j/Makefile.am
@@ -1,5 +1,6 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
-lib_LTLIBRARIES = liblttng-ust-log4j-jni.la
+jnilib_LTLIBRARIES = liblttng-ust-log4j-jni.la
+jnilibdir = $(libdir)/jni
 liblttng_ust_log4j_jni_la_SOURCES = lttng_ust_log4j.c \
 				  lttng_ust_log4j.h
 
--- a/liblttng-ust-java-agent/java/Makefile.am
+++ b/liblttng-ust-java-agent/java/Makefile.am
@@ -1,4 +1,5 @@
 JAVAROOT = .
+AM_JAVACFLAGS = -source 1.6 -target 1.6
 
 pkgpath = org/lttng/ust/agent
 pkgpath_old = org/lttng/ust/jul
