Description: javah doesn't generate class files
Author: Michael Jeanson <mjeanson@ubuntu.com>

--- a/liblttng-ust-java/Makefile.am
+++ b/liblttng-ust-java/Makefile.am
@@ -22,6 +22,7 @@
 org_lttng_ust_LTTngUst.h: jni-header.stamp
 
 jni-header.stamp: $(dist_noinst_JAVA)
+	$(JAVAC) -classpath $(srcdir) $(JAVACFLAGS) $(pkgpath)/LTTngUst.java
 	$(JAVAH) -classpath $(srcdir) $(JAVAHFLAGS) org.lttng.ust.LTTngUst && \
 	echo "JNI header generated" > jni-header.stamp
 
